### ISSUE TITLE
Different "npm install" behavior based on LPAR

### ISSUE THREAD

**Issue by eamondawes** [ibmruntimes/node#98]
Running “npm install” with the same package and package-lock files results in different behavior depending on the LPAR. The install will succeed on some LPARs and fail on others. The LPARs are within the same PLEX and use shared DASD and HFS filesystems.  All environment settings are the same.

This is the package.json
[package.json.txt](https://github.com/ibmruntimes/node/files/1395380/package.json.txt)

This is the error message displayed when "npm install" fails
[install.txt](https://github.com/ibmruntimes/node/files/1395381/install.txt)

This is the package-lock.json from a successful installation
[package-lock.json.txt](https://github.com/ibmruntimes/node/files/1395384/package-lock.json.txt)

zOS 2.01
node v6.11.2
npm v5.4.2

Thank you!

**Comment by mmallick-ca** [comment]
@eamondawes is the failure consistent on specific LPARS? 

Looking at the message in install.txt it seems some temporary package.json file that is created when installing a dependency was not found.  It could be that file could not be created because there was not enough disk space?Could you share the verbose output from a successful and a failed install(npm install --verbose). 

On our development machines I can not reproduce the problem ( all though I seem to run into some python related issues with a post-install script for one of the packages).

**Comment by jBarz** [comment]
Hi Eamon, we managed to get hold of a parallel sysplex system and tried `npm install` using your package.json and it seemed to work fine.
I installed node on my home directory (Client=NO)
I then moved the whole workspace over to /tmp (CLIENT=NO but different owner) and then I got an error with `npm install`
```
fs.js:999
  binding.lstat(pathModule._makeLong(path), statValues);
          ^
Error: ENOENT: no such file or directory, lstat '/SA0D/tmp� '
```

Then I upgraded node to the latest PTF UI51561 (nodejs v6.11.4) and it worked fine.
So that issue is resolved in the latest PTF.

It might be the case that your issue and the above issue are related. So could you apply that PTF 
and test again?

**Comment by jBarz** [comment]
We found a sysplex aware zfs mount with Client=Y.
Then we Installed node on that zfs. 
Then created a directory on that mount and copied over your package.json and ran `npm install`.
That ran successfully as well.

I used node 6.11.2 which is the same version that you are using.

**Comment by jBarz** [comment]
Fixed after upgrading to 6.11.4

**Comment by ghost** [comment]
Hi @jBarz  - as you know this issue was not resolved via 6.11.4 as we thought.  Please reopen this issue.  Thanks.

**Comment by mmallick-ca** [comment]
Based on our debugging session today @eamondawes and @rocketjon we seem to have made some progress in narrowing down the reason that the failure occurs. Given that we are in agreement that the error has something to do with the filesystem and there isn't anything wrong in the Node runtime, I am going to close this issue.
